neqo-transport/src/connection/mod.rs

```diff
@@ -1931,18 +1931,18 @@ impl Connection {
         grease_quic_bit: bool,
     ) -> (PacketType, PacketBuilder) {
         let pt = PacketType::from(cspace);
+        let dcid = path
+            .remote_cid()
+            .map_or(Vec::new(), |id| id.as_ref().to_vec());
```

Comment:
You shouldn't need to allocate a new `Vec` here.  I would suggest changing the signature of `PacketBuilder::short`/`PacketBuilder::long` instead so that they take `Option<impl AsRef<[u8]>`.  You can call on that with `if let Some(...)` or `dcid.map_or(&[], AsRef::as_ref)` or `dcid.map_or(0, |d| d.as_ref().len())` as necessary.